packages/downloads/src/portal/portal-request-download-metadata.ts

```diff
@@ -67,30 +81,61 @@ export function portalRequestDownloadMetadata(
     });
 }
 
-function fetchLastEditDate(params: any): Promise<number | undefined> {
-  const { datasetId, url, type, modified, authentication } = params;
-
-  const layerId = datasetId.split("_")[1] || 0;
+function fetchCacheSearchMetadata(params: any): Promise<ICacheSearchMetadata> {
+  const { format, layerId, url, type, modified, authentication } = params;
 
   if (type !== "Feature Service" && type !== "Map Service") {
-    return Promise.resolve(modified);
+    return Promise.resolve({
+      modified,
+      format: getSearchFormat(format, false)
+    });
   }
 
-  return getLayer({
-    url: `${url}/${layerId}`,
-    authentication
-  }).then((layer: any) => {
-    const editingInfo: any | null = layer.editingInfo;
-    return editingInfo ? editingInfo.lastEditDate : undefined;
-  });
+  return getService({ url, authentication }).then(
+    (response: IFeatureServiceDefinition) => {
+      const layers: ILayerDefinition[] = response.layers;
+      const multilayer = isMultilayerRequest(layerId, layers);
+      return {
+        format: getSearchFormat(format, multilayer),
+        modified: extractLastEditDate(layers)
+      };
+    }
+  );
+}
+
+function isMultilayerRequest(layerId: string, layers: ILayerDefinition[]) {
+  return layerId === undefined && layers.length > 1;
+}
+
+function extractLastEditDate(layers: ILayerDefinition[]) {
+  const result = layers
+    .map((layer: ILayerDefinition) => {
+      const { editingInfo: { lastEditDate } = {} } = layer;
+      return lastEditDate;
+    })
+    .filter(timestamp => {
+      return timestamp !== undefined;
+    })
+    .sort((a, b) => {
+      return b - a;
+    });
+  return result[0];
+}
+
+function getSearchFormat(format: DownloadFormat, multilayer: boolean) {
```

Comment:
Generally considered an anti-pattern (https://martinfowler.com/bliki/FlagArgument.html), why not just be explicit?

Reply:
Eliminated the flag param.